opm/autodiff/SimulatorFullyImplicitBlackoilEbos.hpp

```diff
@@ -198,6 +209,11 @@ class SimulatorFullyImplicitBlackoilEbos
                                     prev_well_state,
                                     restorefilename,
                                     desiredRestoreStep );
+            initHydroCarbonState(state, phaseUsage_, Opm::UgGridHelpers::numCells(grid()), has_disgas_, has_vapoil_);
+            initHysteresisParams(state);
+            // communicate the restart solution to ebos
+            convertInput(0, state, ebosSimulator_);
```

Comment:
The convertInput() and cache invalidation calls seem to be repeated. Would it be wrong to collect them below?